### DISCUSSION TITLE
Beginner tutorial?

### DISCUSSION THREAD

**Discussion by sensn** [ObsidianPublisher/obsidian-github-publisher#130] [Q&A — Plugin] [answered]
Hello, I am new to obsidian and github pages and I have some questions about setting up and deploying my notes to GH pages.
I followed the quick tutorial and finally got some working repos, but I also have some testrepos which do not deploy correctly or do not deploy at all(although they show success in GH actions).

One thing i noticed that I had to use "http://" instead of "https://".

But what i really do not understand is what " Fixed folder", "Obsidian Path" and "frontmatter" really do and which one to use. 
 
My first testrepos work but now i cannot reproduce the step to get a working page.

https://sensn.github.io/testvault/

This works, but how would I create a landing page instead of 404?

https://sensn.github.io/vault6/

This one doesn't work correctly, although its the same files, I am not sure whats wrong.

Maybe a verry basic step-by-steo beginnee tutorial would be great!
Ve

**Comment by Mara-Li** [comment]
Documentation already explain what do each folder option.

Also, you can't publish to multiple different repo. So yes, one or multiple repo can be broken based on your configuration

**Comment by h7b** [comment]
> But what i really do not understand is what " Fixed folder", "Obsidian Path" and "frontmatter" really do and which one to use.

-> i suggest to use `Obsidian Path`, since this option allow the GitHub Publisher plugin to replicate the folder structure of your Obsidian vault into the mkdocs published site. Here is the [link to the plugin's documentation](https://obsidian-publisher.netlify.app/github%20publisher/settings/upload/) which explains the "folder reception settings" and "frontmatter key"

> Maybe a verry basic step-by-steo beginnee tutorial would be great!

-> As you may guess correctly that Mara is an experienced developer, so something which is obvious to her would be not easy for us to understand. That's the blind spot in point of view. Further, you have a better chance to receive an appropriate answer if you can describe your question carefully, and explain what you did that lead to that question in details.

> This works, but how would I create a landing page instead of 404?

-> You can create a landing page by create a note named `index.md` at the root of your Obsidian vault. When you publish, this file will be served as the home page

**Reply by Mara-Li** [reply]
Thanks! I was occupied so can't link to the documentations :).
And yes, as I develop everything in solo', I also created the tutorial, but something maybe not right with the tuto, the screen or something. It's difficult to maintain everything ^^"

**Reply by sensn** [reply]
Thanks for the info, I'll keep trying. One more question: I have "transfer attachments" enabled 
,so I'd expect the plugin to upload an image if i have a "![[image.jpg]] link in my note. But it doen't seem to work that way. What am missing? Thanks for your help.do Ihave to create a frontmatter for my images too? What if i have audio/midi files?

**Reply by Mara-Li** [reply]
No, you can't create frontmatter for attachments!
Every attachments supported by Obsidian can be send to the repo.

Here my settings for my own mkdocs site : 

```json
{
  "github": {
    "branch": "main",
    "automaticallyMergePR": true,
    "api": {
      "tiersForApi": "Github Free/Pro/Team (default)",
      "hostname": ""
    },
    "worflow": {
      "workflowName": "",
      "customCommitMsg": "[skip netlify] PUBLISHER"
    }
  },
  "upload": {
    "behavior": "yaml",
    "subFolder": "",
    "defaultName": "docs/hidden",
    "rootFolder": "docs",
    "yamlFolderKey": "category",
    "frontmatterTitle": {
      "enable": true,
      "key": "name"
    },
    "replaceTitle": [],
    "autoclean": {
      "enable": true,
      "excluded": [
        "docs/assets/js",
        "docs/assets/meta",
        "docs/assets/css",
        "tags.md",
        "graph.md"
      ]
    },
    "folderNote": {
      "enable": true,
      "rename": "index.md"
    },
    "metadataExtractorPath": "",
    "replacePath": []
  },
  "conversion": {
    "hardbreak": false,
    "dataview": false,
    "censorText": [],
    "tags": {
      "inline": true,
      "exclude": [],
      "fields": []
    },
    "links": {
      "internal": true,
      "unshared": false,
      "wiki": false
    }
  },
  "embed": {
    "attachments": true,
    "keySendFile": [
      "image"
    ],
    "notes": true,
    "folder": "docs/assets/img"
  },
  "plugin": {
    "shareKey": "share",
    "fileMenu": true,
    "editorMenu": true,
    "excludedFolder": [],
    "externalShare": false,
    "copyLink": {
      "enable": true,
      "links": "",
      "removePart": [
        "docs/",
        ".md",
        "index"
      ],
      "addCmd": true
    },
    "noticeError": false,
    "displayModalRepoEditing": false
  }
}
```

**Reply by sensn** [reply]
Thanks, for sharing !

Meanwhile I started all over again with a new vault and new repo. Didn't know that plugins are loaded per vault in Obsidian. It all makes a little bit more sense now.

**Reply by sensn** [reply]
@Lisandra-dev 
Hello,
I have another question. I can't get the logo.png to work, and i also tried the banner css, but couldn't get it right.
I use the GH-template. In the readme it says:
`To edit the logo and favicon, first put the chosen files in the assets/logo directory, and then change logo and favicon:

logo: assets/meta/logo_name.png
favicon: assets/meta/favicon.png`

is this the docs/assets folder? because there is no assets/logo folder. i created one " docs/assets/logo/logo.png" but that didn't work. also tried "docs/assets/meta/logo.png".
I also set that in mkdocs.yml  "theme:logo: ..."
But the logo isn't shown.

For the banner i tried uncommenting the first block in css/customization.css and used a "https://" link to a online jpg. Could you help me please? mybe with y short example?

BTW, I created m own workflow now, using "Obsidian Git" plugin for pushing to my git repo. From there i set up an action which pushes the conntent of the "docs"  folder to the "docs" folder of the repo generated from the GH_template repo.
This way i can sync with push/pull from several devices and have the site auto- generated. seems to work so far.
Cheers, sensn

**Reply by Mara-Li** [reply, marked as answer]
First of all, I don't advice you to use your mkdocs repo with your github repo, as files are changed by my plugin.

For the template, there is a typo. You need to use the folder `docs/assets/meta`. You can check the mkdocs.yml here : https://github.com/ObsidianPublisher/obsidian-mkdocs-publisher-docs/blob/main/mkdocs.yml

Here is my banner : 
```css
.md-header,
.md-nav--primary .md-nav__title[for="__drawer"] {
  background: var(--md-primary-fg-color)
    url(https://images.unsplash.com/photo-1520157646479-22f25713523d?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1074&q=80)
    left center/cover no-repeat;
}
```

**Reply by sensn** [reply]
Oh, nice, now both the logo and the banner work! Thank you!

I have a repo which is a mirror of my obsidian vault. from this repo i push automatically to another repo - which is generated from https://github.com/ObsidianPublisher/publisher-template-gh-pages.

Most of the time it works. sometimes i get 404 after buiding. but i don't know why. it shows enviroment gh_pages inactive. although bulid and deploy says success.

I seem to have problems embedding pdfs, and png. but jpg works fine.
Do embedded pdfs work with your plugin?

Here is a link to my testing repo, just for fun:

https://github.com/sensn/DigitalGarden/
https://sensn.github.io/DigitalGarden/

Sometimes its up sometimes down. I don't get it why it fails sometimes, if its me or the server.

**Reply by Mara-Li** [reply]
Embedding pdf... Dunno, to be honest. Doesn't tried. I don't think so, you need to use another mkdocs plugin for that, i think.
I don't have any problem with png or jpeg, so I don't understand why you get 404 ?
